### PR TITLE
Ports Hydrogen and it's counterparts from TG

### PR DESCRIPTION
**Pull request by utilbird**
tgstation/tgstation#50510

## About The Pull Request

Adds hydrogen, a fun, hot gas. Also makes plasma act more consistently with it's implied hydrocarbon-ish chemistry.

Fusion now uses Hydrogen as it's second main component instead of Carbon Dioxide

bunch more stuff like electrolyzers, ammonia crystals and metallic hydrogen. Read original PR for a better description

## Why It's Good For The Game

Read original pr. 

## Changelog
:cl:
add: Adds Hydrogen gas
add: Adds Metallic Hydrogen, gas reaction product
add: Adds Ammonia Crystals, gas reaction product
add: Adds Metallic Hydrogen golems
add: Adds the electrolyzer, turn water into hydrogen and oxygen with this one simple trick
del: Removed unobtanium as a placeholder gas
tweak: Hydrogen is now used instead of CO2 in fusion
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
finally, something to do as an atmos tech

**Comment by Sinestia** [comment]
Did this uhhh... work without extools changes?


 Unless you just did that privately and will submit a PR when this is approved :shrug:


edit: wait you literally just overrided unobtanium lmao

**Comment by ike709** [comment]
rip unobtainium

**Comment by BrumalLake** [comment]
> 
> 
> rip unobtainium

noo i cant get unobtanium anymore 😭😭😭

**Comment by utilbird** [comment]
unobtanium existed as a placeholder gas so we hit the 14 gas extools define. There's been two PRs that remove the limit but both have been shot down due to EoL:tm:. We must kiss unobtanium goodbye :pensive:

File: code/game/machinery/electrolyzer.dm [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/electrolyzer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
+										datum/tgui/master_ui = null, datum/ui_state/state = GLOB.physical_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "Electrolyzer", name, master_ui, state)
+		ui.open()
```
Comment (Sarchutar) [inline code/game/machinery/electrolyzer.dm]:
```suggestion
/obj/machinery/electrolyzer/ui_state(mob/user)
	return GLOB.physical_state

/obj/machinery/electrolyzer/ui_interact(mob/user, datum/tgui/ui)
	ui = SStgui.try_update_ui(user, src, ui)
	if(!ui)
		ui = new(user, src, "Electrolyzer", name)
		ui.open()
```
(from tg)

File: tgui/packages/tgui/interfaces/Electrolyzer.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Fragment } from 'inferno';
+import { useBackend } from '../backend';
+import { Box, Button, LabeledList, NumberInput, ProgressBar, Section } from '../components';
+import { Window } from '../layouts';
+
+export const Electrolyzer = (props, context) => {
+  const { act, data } = useBackend(context);
+  return (
+    <Window>
```
Comment (Sarchutar) [inline tgui/packages/tgui/interfaces/Electrolyzer.js]:
```suggestion
    <Window
      width={400}
      height={305}>
```
(same as other one)

**Comment by PowerfulBacon** [comment]
We should add copium instead

**Comment by BrumalLake** [comment]
I'm pretty sure ghilker removed ammonia crystals cause they didnt do anything special at all and were a bloat reaction, do we really need them

**Comment by utilbird** [comment]
bloat for now:tm:

**Comment by mat-lap** [comment]
yesy eys yeysye ysyeysysy s yes atmos shit

**Comment by BrumalLake** [comment]
https://github.com/tgstation/tgstation/pull/57591

**Comment by PowerfulBacon** [comment]
Reopen if this works with auxmos and doesnt conflict with half the codebase.

**Pull request closed without merging**